src/Microsoft.Data/BaseColumn.cs

```diff
@@ -34,8 +35,23 @@ protected set
 
         public string Name;
 
-        public virtual object this[long rowIndex] { get { throw new NotImplementedException(); } set { throw new NotImplementedException(); } }
+        public readonly Type DataType;
```

Comment:
This should be a property. All these exposed fields should be properties instead - e.g. Name.

In general, values are exposed as properties in C#, not fields. Only on rare occasions are fields exposed outside of a class or struct.

Reply:
I actually find it tough to think of any situation where we would expose a field from a class.